components/base/sensorcontrolled/spin.go

```diff
@@ -26,7 +23,7 @@ func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, ex
 		sb.logger.CWarnf(ctx, "control parameters not configured, using %v's spin method", sb.controlledBase.Name().ShortName())
 		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, extra)
 	}
-	if sb.orientation == nil {
+	if sb.orientation == nil && sb.compassHeading == nil {
```

Comment:
explicitly checking for both because the `headingFunc`'s default will return an angle of 0